src/Translator/src/Twig/TranslatorExtension.php

```diff
@@ -0,0 +1,59 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\UX\Translator\Twig;
+
+use Symfony\Component\Translation\MessageCatalogueInterface;
+use Symfony\Component\Translation\TranslatorBagInterface;
+use Twig\Extension\AbstractExtension;
+use Twig\TwigFunction;
+
+/**
+ * @author Hugo Alliaume <hugo@alliau.me>
+ */
+final class TranslatorExtension extends AbstractExtension
+{
+    public function __construct(
+        private TranslatorBagInterface $translatorBag
+    ) {
+    }
+
+    public function getFunctions(): array
+    {
+        return [
+            new TwigFunction('initialize_js_translator', [$this, 'initializeJsTranslator'], ['is_safe' => ['all']]),
+        ];
+    }
+
+    public function initializeJsTranslator(string $locale = null): string
+    {
+        $locale ??= class_exists(\Locale::class) ? \Locale::getDefault() : 'en';
+
+        return sprintf(
+            'window.__symfony_ux_translator = %s;',
+            json_encode([
+                'locale' => $locale,
+                'locales_fallbacks' => (object) $this->getLocalesFallbacks(...$this->translatorBag->getCatalogues()),
```

Comment:
As stated elsewhere, `locales_fallbacks` isn't dependent on the passed locale (so unless I'm misunderstanding, it would be same for every request). Can we move this to the `cache:warmup` process?